app/javascript/dashboard/store/modules/conversations/getters.js

```diff
@@ -9,10 +11,14 @@ export const getSelectedChatConversation = ({
 
 // getters
 const getters = {
-  getAllConversations: ({ allConversations }) =>
-    allConversations.sort(
+  getAllConversations: ({ allConversations, chatSortFilter }) => {
+    if (chatSortFilter === SORT_BY_TYPE.CREATED_AT) {
+      return allConversations.sort((a, b) => b.created_at - a.created_at);
+    }
+    return allConversations.sort(
```

Comment:
The sorting based on the last message, `created_at` is intentional here. If we use conversation `last_activity_at` to sort the chats, the incoming message will not push the conversation to the top of the list since the `last_activity_at` will not be updated on the `message_created` event. 
The solution is to update the conversation `last_activity_at` on the `message_created` event. This needs some changes in the backend.

@sojan-official @pranavrajs